src/Voron/Data/Containers/Container.cs

```diff
@@ -316,28 +308,196 @@ public string Dump()
+                        switch (sizeOfElement)
+                        {
+                            case 0:
+                                usedItems--;
+                                continue;
+                            case 30:
+                                size += *(pagePtr + offsets.GetElement(i));
+                                break;
+                            case 31:
+                                size += *(ushort*)(pagePtr + offsets.GetElement(i));
+                                continue;
+                            default:
+                                Debug.Assert(sizeOfElement is < 30 and > 0);
+                                break;
+                        }
+                    }
+                }
+            }
+            
+            if (AdvInstructionSet.IsAcceleratedVector256)
+            {
+                var N = Vector256<ushort>.Count;
+                for (; index + N <= numberOfOffsets; index += N)
+                {
+                    var metadataItems = Vector256.Load((ushort*)_page.DataPointer + index);
+                    var offsets = Vector256.BitwiseAnd(
+                        left: Vector256.ShiftRightLogical(metadataItems, ItemMetadata.OffsetShift), 
+                        right: Vector256.Create(ItemMetadata.OffsetMask));
+                    var sizes = Vector256.BitwiseAnd(
+                        left: metadataItems, 
+                        right: Vector256.Create(ItemMetadata.SizeMask));
+
+                    for (var i = 0; i < N; i++)
+                    {
+                        var sizeOfElement = sizes.GetElement(i);
+                        switch (sizeOfElement)
+                        {
+                            case 0:
+                                usedItems--;
+                                continue;
+                            case 30:
+                                size += *(pagePtr + offsets.GetElement(i));
+                                break;
+                            case 31:
+                                size += *(ushort*)(pagePtr + offsets.GetElement(i));
+                                continue;
+                            default:
+                                Debug.Assert(sizeOfElement is < 30 and > 0);
+                                break;
+                        }
+                    }
+                }
+            }
+            
+            if (AdvInstructionSet.IsAcceleratedVector128)
+            {
+                var N = Vector256<ushort>.Count;
+                for (; index + N <= numberOfOffsets; index += N)
+                {
+                    var metadataItems = Vector128.Load((ushort*)_page.DataPointer + index);
+                    var offsets = Vector128.BitwiseAnd(
+                        left: Vector128.ShiftRightLogical(metadataItems, ItemMetadata.OffsetShift), 
+                        right: Vector128.Create(ItemMetadata.OffsetMask));
```

Comment:
This is easier to write & understand

```suggestion
                    var offsets = (metadataItems >> ItemMetadata.OffsetShift) & 
                                          Vector128.Create(ItemMetadata.OffsetMask)
```